serverless/executor/src/node_handler.rs

```
@@ -171,13 +216,24 @@ pub async fn inject_mutable_config(
     let pos = ws_rpc_url.rfind('/').unwrap();
     ws_rpc_url.truncate(pos + 1);
     ws_rpc_url.push_str(mutable_config.ws_api_key.as_str());
-    *mutable_params_injected_guard = true;
 
-    (StatusCode::OK, "Mutable params configured!\n").into_response()
+    if secret_store_response.0 == reqwest::StatusCode::OK {
```

can you handle it just after the http call to secret store? Any specific reason to check it at the end?

---

Even if setting mutable params in the store fails, we should still proceed to try setting it for the executor. The last check is just to mark if mutable params have been injected in both store and executor or not which will allow us to export registration message for the enclave.

---

1. If we return http error at L209, i.e. just after secret store api call failure, what problem will it cause?
2. If secret store api fails, then user has to set mutable config again, right?

---

1. There is no problem, but we can return the error after setting the executor mutable params also if possible since mutable params for both are independent and can be needed to set again for them separately. 
2. Yes right. If secret store api fails then we'll return the error in the end and won't set mutable_params_injected to true if not already.

---

why do you want to set the mutable param in executor even if secret store api fails? Given, user has to call the api again.

Also, you are creating inconsistency in the executor state. That mutable_params_injected_guard is false but the ws_api, nonce_to_send and http_rpc_client has been set.

I want to keep this invariability.